src/retryable.rs

```diff
@@ -0,0 +1,128 @@
+use std::{future::Future, time::Duration};
+
+#[cfg(feature = "async")]
+use async_std::task::sleep;
+#[cfg(feature = "async")]
+use reqwest::{RequestBuilder, Response as ReqwestResponse, StatusCode};
+
+#[cfg(feature = "blocking")]
+use ureq::{Error as UReqError, Request, Response as UReqResponse};
+
+const SECOND: u64 = 1_000;
+const BASE_BACKOFF_MS: u64 = 5_000;
+const DEFAULT_MAX_RETRIES: u32 = 5;
+
+pub const RETRY_TOO_MANY_REQUEST_ONLY: [StatusCode; 1] = [StatusCode::TOO_MANY_REQUESTS];
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After, maybe we should also add 503 `SERVICE_UNAVAILABLE` to the list?

Reply:
Done in [1c64b9b](https://github.com/bitcoindevkit/rust-esplora-client/pull/71/commits/1c64b9bdfe0d3cc17c6630db2dfdb1236b9b11ae)